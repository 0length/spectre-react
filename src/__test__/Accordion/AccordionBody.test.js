import React from 'react'
import { shallow } from 'enzyme'

import { AccordionBody } from '../..'

describe('AccordionBody', () => {
  it('Should render default .accordion-body markup', () => {
    const wrapper = shallow(<AccordionBody />)

    expect(wrapper.type()).toBe('div')
    expect(wrapper.hasClass('accordion-body')).toBe(true)
  })

  it('Should render children', () => {
    const wrapper = shallow(
      <AccordionBody>
        <p>hello world</p>
      </AccordionBody>
    )

    expect(wrapper.children().length).toBe(1)
    expect(wrapper.find('p').length).toBe(1)
  })

  it('Should pass classnames', () => {
    const wrapper = shallow(<AccordionBody className='extra' />)

    expect(wrapper.hasClass('extra')).toBe(true)
    expect(wrapper.hasClass('accordion-body')).toBe(true)
  })

  it('Should render as a custom element', () => {
    const wrapper = shallow(<AccordionBody renderAs='main'>custom</AccordionBody>)

    expect(wrapper.text()).toBe('custom')
    expect(wrapper.hasClass('accordion-body')).toBe(true)
    expect(wrapper.type()).toBe('main')
  })

  it('Should match the snapshot', () => {
    const wrapper = shallow(<AccordionBody />)

    expect(wrapper).toMatchSnapshot()
  })
})
