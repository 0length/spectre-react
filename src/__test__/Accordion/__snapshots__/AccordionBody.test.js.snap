// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordionBody Should match the snapshot 1`] = `
<div
  className="accordion-body"
/>
`;
